From 0ec7b4ae5e4a08f7bdbcb6f9eb963212bebfa43b Mon Sep 17 00:00:00 2001
From: Yonghong Song <ysong@broadcom.com>
Date: Tue, 25 Jun 2013 13:33:34 -0700
Subject: [PATCH 0371/1532] HEv2: pktmem: add packet memory attributes to
 xlp9xx-linux/xlp9xx-linux-guest dts

[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/boot/dts/xlp9xx-linux-guest.dts | 27 ++++++++++++++++++++++++++-
 arch/mips/boot/dts/xlp9xx-linux.dts       | 10 ++++++++++
 2 files changed, 36 insertions(+), 1 deletion(-)

diff --git a/arch/mips/boot/dts/xlp9xx-linux-guest.dts b/arch/mips/boot/dts/xlp9xx-linux-guest.dts
index c774ab8..6f9869c 100644
--- a/arch/mips/boot/dts/xlp9xx-linux-guest.dts
+++ b/arch/mips/boot/dts/xlp9xx-linux-guest.dts
@@ -15,8 +15,33 @@
                         0 0x20000000 0 0x10000000>;     // 256M at 512M
         };
 
+        /* memory range for HEv2 pkt mem. Only one contig memory per node supported */
+	/* size 0 will be ignored. */
+        pktmem {
+                device_type = "pktmem";
+
+                /* format: node number, 64-bit addr, 64-bit size */
+                reg = <0 0x0 0x30000000 0x0 0x0000000
+                       1 0x0 0x0 0x0 0x0
+                       2 0x0 0x0 0x0 0x0
+                       3 0x0 0x0 0x0 0x0>;
+        };
+
+	doms {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		dom@1 {
+			device_type = "domain";
+			os = "linux";
+
+			cpu {
+			};
+		};
+	};
+
 	chosen {
-		netlogic,cpumask = <0xff>; /* This is only for simulator and temporary. should be fixed later */
+		netlogic,cpumask = <0x1>;
 		bootargs = " console=ttyS0,115200  rhash_entries=1048576";
 	};
 
diff --git a/arch/mips/boot/dts/xlp9xx-linux.dts b/arch/mips/boot/dts/xlp9xx-linux.dts
index dbba6af..50cf9ab 100644
--- a/arch/mips/boot/dts/xlp9xx-linux.dts
+++ b/arch/mips/boot/dts/xlp9xx-linux.dts
@@ -15,6 +15,16 @@
                         0 0x20000000 0 0xa0000000>;     // 2560M at 512M
         };
 
+        /* memory range for HEv2 pkt mem. Only one contig memory per node supported */
+        pktmem {
+                device_type = "pktmem";
+
+                /* format: node number, 64-bit addr, 64-bit size */
+                reg = <0 0x0 0xe0000000 0x0 0x20000000
+                       1 0x0 0x0 0x0 0x0
+                       2 0x0 0x0 0x0 0x0
+                       3 0x0 0x0 0x0 0x0>;
+        };
 
 	hypervisor {
 		hypervisor-name = "Xen";
-- 
1.9.1

